### PR TITLE
Ban package:path from Flutter Tools

### PR DESCRIPTION
**Pull request by goderbauer**
Use `fs.path` instead for cross-platform interoperability.

### CODE REVIEW

**Comment by Hixie** [comment]
I thought package:path was supposed to be cross-platform?

File: packages/flutter_tools/lib/src/commands/drive.dart [hunk 1]
```diff
@@ -1,358 +1,356 @@
...
-}
-
-Future<bool> _stopApp(DriveCommand command) async {
-  printTrace('Stopping application.');
-  ApplicationPackage package = command.applicationPackages.getPackageForPlatform(command.device.platform);
-  bool stopped = await command.device.stopApp(package);
-  await command._deviceLogSubscription?.cancel();
-  return stopped;
-}
+// Copyright 2016 The Chromium Authors. All rights reserved.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/drive.dart]: something really bad happened here...

File: packages/flutter_tools/test/forbid_package_path_test.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+              fail("${file.path} imports 'package:path/path.dart'; use 'fs.path' instead");
+            }
+          }
+        }
+      );
+    }
+  });
+}
+
+bool _isDartFile(FileSystemEntity entity) =>
```
Comment (Hixie) [inline packages/flutter_tools/test/forbid_package_path_test.dart]: nit: one-line or { return ... }

File: packages/flutter_tools/test/forbid_package_path_test.dart [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  test('no unauthorized imports of package:path', () {
+    for (String path in <String>['lib', 'bin', 'test']) {
+      fs.directory(path)
+        .listSync(recursive: true)
+        .where(_isDartFile)
+        .map(_asFile)
+        .forEach((File file) {
+          for (String line in file.readAsLinesSync()) {
+            if (line.startsWith(new RegExp('import.*package:path/path.dart')) &&
+                !line.contains('ignore: package_path_import')) {
```
Comment (Hixie) [inline packages/flutter_tools/test/forbid_package_path_test.dart]: `ignore: package_path_import` doesn't seem to be used anywhere, so I would remove this ability

**Comment by Hixie** [comment]
A bunch of the files have changes on every line, which seems bad. I'm guessing LF->CRLF?

Should we add a test that no files contain U+000D?

File: packages/flutter_tools/lib/src/commands/drive.dart (hunk 1 above)
Comment (goderbauer) [inline packages/flutter_tools/lib/src/commands/drive.dart]: Line endings...

**Comment by goderbauer** [comment]
package:path is cross platform, but it doesn't work well with Todd's record-replay filesystem in tests. For that to work, you have to use the filesystem's implementation of path.

Yes, we need some kind of a check to prevent Windows line endings. Unfortunately, right now I only see them when I upload stuff to github :(. Will file an issue.

**Comment by goderbauer** [comment]
Filed: https://github.com/flutter/flutter/issues/8121

**Comment by Hixie** [comment]
LGTM

**Comment by tvolkert** [comment]
Yeah, package:path is cross platform inasmuch as it detects your current platform and uses path separators for that platform.  But when you're using an in-memory file system or replaying a recording that was recorded on another platform, you explicitly don't want your platform's behavior.  Thus It's best when using FileSystem to use the path context retrieved from the file system object.

**Comment by tvolkert** [comment]
Incidentally fs.path returns a path.Context from package:path

**Comment by tvolkert** [comment]
I can't help but think that we may want to define the following in file_system.dart:

```
path.Context get path => fs.path;
```

Just to keep from updating all these files....

**Comment by goderbauer** [comment]
Hm, but we would still be modifying all those files to remove the import. Doesn't seem to make that much of a difference especially since the work is done now...

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**